crates/sui-bridge/src/action_executor.rs

```diff
@@ -230,8 +230,17 @@ where
         >,
     ) {
         let BridgeActionExecutionWrapper(action, attempt_times) = action;
+
+        // Only token transfer action should reach here
+        match &action {
+            BridgeAction::SuiToEthBridgeAction(_) | BridgeAction::EthToSuiBridgeAction(_) => (),
```

Comment:
If not too late, consider adding a secondary enum for token transfer action

Reply:
It's not too late (based on my estimation on the work), but I don't think it will give us much benefits beyond a couple of match arms.
A potential good chance to refactor these actions may be when we introduce more chain support, at which time hopefully we have better ideas how to group them